Project.toml

```
@@ -8,7 +8,7 @@ FixedPointNumbers = "53c48c17-4a7d-5ca2-90c5-79b7896eea93"
 Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
 
 [compat]
-ColorTypes = "0.10, 0.11"
+ColorTypes = "0.12"
```

```suggestion
ColorTypes = "0.11.5, 0.12"
```
See https://github.com/JuliaGraphics/ColorTypes.jl/pull/296

---

This is not good. This should be "backported".

---

Could you expand on this?

---

You're saying that because ColorTypes v0.12 is not yet released, it shouldn't be included in compat bounds until release (so the compat bounds would be just v0.11.5)?

---

Will this be released as Colors v0.13?
I was under the impression that it would be backported to Colors v0.12.
In other words, I was wondering if an `isdefined` switch is needed instead of changing the `compat`.

---

Since it is not practical to write many `isdefined` switches, I think it would be practical to define dummy `Oklab` and `Oklch` in Colors.jl.
Of course, that is not a good practice by any means.

---

I don't feel backporting makes any sense? This depends on a new type in a to be released ColorTypes version (which does not yet exist as far as I know) and only adds functionality related to that type. A new ColorTypes release is blocking to merging this code, and that seems fine to me.

Of course, the compat should be updated at that time, but we can't do that yet as the version does not exist. We should leave this comment open to remind of that fact.

---

I have no strong opinions about backporting.
If I have to say, I do not want to backport because it is too much trouble.

However, since `Oklab` and `Oklch` were backported to ColorTypes.jl v0.11, I think there is at least a need for backporting.

---

That being said, can't this just go into a v0.13? Or is there any particular reason that we would want to stick with v0.12?

---

There is no absolute reason.
However, FixedPointNumbers.jl, ColorTypes.jl, Colors.jl, and ColorVectorSpace.jl are used in many packages, and ”unplanned” minor version increases will bring chaos.

It would be nice if the maintainers of each downstream package would carefully check the compatibility and update their `[compat]`.

---

Adapted to conform to later discussion in the main pull request discussion. Marking as resolved for now.